clientbase/ops.go

```diff
@@ -41,6 +41,9 @@ func (a *APIOperations) DoDelete(url string) error {
 	defer func() {
 		io.Copy(ioutil.Discard, resp.Body)
 		resp.Body.Close()
+		if tr, ok := a.Client.Transport.(*http.Transport); ok {
```

Comment:
This seems excessive.  If you create an http.Client you have two options.  1) cache the object and reuse it 2) use it once and call CloseIdleConnections().  So I can see that in NewAPIClient we should call CloseIdleConnections() because we create single use http Clients.  But in this method we shouldn't be closing idle connection per HTTP request.  The only way this is a problem is if somebody is calling NewAPIClient and then discarding the rancher API client.  For a fix for that the caller should be responsible for closing the idle connections.  So we can add a method to the rancher API client CloseIdleConnections() that a user can call.